lazy.md

```diff
@@ -13,3 +14,1317 @@ author:
+the corresponding allocator if present. For example:
+
+    struct allocator_aware_context {
+        using allocator_type = std::pmr::polymorphic_allocator<std::byte>;
+    };
+
+    template <typename...A>
+    ex::lazy<int, allocator_aware_context> fun(int value, A&&...) {
+        co_return value;
+    }
+    
+    int main() {
+        // Use the coroutine without passing an allocator:
+        ex::sync_wait(fun(17));
+
+        // Use the coroutine with passing an allocator:
+        using allocator_type = std::pmr::polymorphic_alloctor<std::byte>;
+        ex::sync_wait(fun(17, std::allocator_arg, allocator_type()));
+    }
+
+The arguments passed when creating the coroutine are made available
+to an `operator new` of the promise type, i.e., this operator can
+extract the allocator, if any, from the list of parameters and use
+that for the purpose of allocation. The matching `operator delete`
+gets passed only the pointer to release and the originally requested
+`size`. To have access to the correct allocator in `operator delete`
+the allocator either needs to be stateless or a copy needs to be accessible
+via the pointer passed to `operator delete`, e.g., stored at the
+offset `size`.
+
+To avoid any cost introduced by type erasing an allocator
+type as part of the `lazy` definition the expected allocator type
+is obtained from the context argument `C` of `lazy<T, C>`:
+
+    using allocator_type = ex::allocator_of_t<C>;
+
+This `using` alias uses `typename C::allocator_type` if present or
+defaults to `std::allocator<std::byte>` otherwise.  This
+`allocator_type` has to be for the type `std::byte` (if necessary
+it is possible to relax that constraint).
+
+The allocator used for the coroutine frame should also be used for
+any other allocators needed for the coroutine itself, e.g., when
+type erasing something needed for its operation (although in most
+cases a small object optimization would be preferable and sufficient).
+Also, the allocator should be made available to child operations
+via the respective receiver's environment using the `get_allocator`
+query. The arguments passed to the coroutine are also available to
+the constructor of the promise type (if there is a matching on) and
+the allocator can be obtained from there:
+
+    struct allocator_aware_context {
+        using allocator_type = pmr::polymorphic_allocator<std::byte>;
+    };
+    fixed_resource<2048> resource;
+
+    ex::sync_wait([](auto&&, auto* resource)
+            -> ex::lazy<void, allocator_aware_context> {
+        auto alloc = co_await ex::read_env(ex::get_allocator);
+        use(alloc);
+    }(allocator_arg, &resource));
+
+## Environment Support
+
+When `co_await`ing child operations these may want to access an
+environment. Ideally, the coroutine would expose the environment
+from the receiver it gets `connect`ed to. Doing so isn't directly
+possible because the coroutine types doesn't know about receiver
```

Comment:
```suggestion
possible because the coroutine types doesn't know about the receiver
```
```suggestion
possible because the coroutine types doesn't know about receiver
```